pref-adjust.lua

```diff
@@ -1,80 +1,165 @@
-    pss_counter = pss_counter + 1
-    -- likes iron, steel (0,8) adam is 25
-    utils.insert_or_update(unit.status.current_soul.preferences, { new = true, type = 0 , item_type = -1 , creature_id = -1 , color_id = -1 , shape_id = -1 , plant_id = -1 , item_subtype = -1 , mattype = 0 , matindex = dfhack.matinfo.find("IRON").index , active = true, prefstring_seed = pss_counter }, 'prefstring_seed')
-    pss_counter = pss_counter + 1
-    utils.insert_or_update(unit.status.current_soul.preferences, { new = true, type = 0 , item_type = -1 , creature_id = -1 , color_id = -1 , shape_id = -1 , plant_id = -1 , item_subtype = -1 , mattype = 0 , matindex = dfhack.matinfo.find("STEEL").index , active = true, prefstring_seed = pss_counter }, 'prefstring_seed')
+    -- Type 2: Prefers to consume drink: (From plump helmets we get dwarven wine)
+    insert_preference(unit,2,{df.item_type.DRINK,"MUSHROOM_HELMET_PLUMP:DRINK"})
+    -- Type 2: Prefers to consume food: (plump helmets, mushrooms)
+    insert_preference(unit,2,{df.item_type.PLANT,"MUSHROOM_HELMET_PLUMP:MUSHROOM"})
+    -- Type 2: Prefers to consume prepared meals: (quarry bush)
+    insert_preference(unit,2,{df.item_type.FOOD,"BUSH_QUARRY"})
 
-    prefcount = #(unit.status.current_soul.preferences)
-    print ("After, unit "..dfhack.TranslateName(dfhack.units.getVisibleName(unit)).." has "..prefcount.." preferences")
+    -- Type 3: Creature detests (TROLL, BIRD_BUZZARD, BIRD_VULTURE, CRUNDLE)
+    insert_preference(unit,3,list_of_creatures.TROLL)
+    insert_preference(unit,3,list_of_creatures.BIRD_BUZZARD)
+    insert_preference(unit,3,list_of_creatures.BIRD_VULTURE)
+    insert_preference(unit,3,list_of_creatures.CRUNDLE)
+end -- end IDEAL profile
 
-end
+if profile == "GOTH" then
+    insert_preference(unit,0,"CREATURE:DWARF:SKIN")
+    insert_preference(unit,4,df.item_type.CORPSE)
+    insert_preference(unit,4,df.item_type.CORPSEPIECE)
+    insert_preference(unit,4,df.item_type.REMAINS)
+    insert_preference(unit,4,df.item_type.COFFIN)
+    insert_preference(unit,7,list_of_colors.BLACK)
+    insert_preference(unit,8,list_of_shapes.CROSS)
+    insert_preference(unit,5,dfhack.matinfo.find("GLUMPRONG").index)
+    insert_preference(unit,2,{df.item_type.DRINK,"WEED_RAT:DRINK"})
+    insert_preference(unit,2,{df.item_type.DRINK,"SLIVER_BARB:DRINK"})
+    insert_preference(unit,2,{df.item_type.PLANT,"TUBER_BLOATED:STRUCTURAL"})
+    insert_preference(unit,3,list_of_creatures.ELF)
+    insert_preference(unit,3,list_of_creatures.HUMAN)
+    insert_preference(unit,3,list_of_creatures.DWARF)
+    if list_of_creatures.DEMON_1 and df.global.world.raws.creatures.all[list_of_creatures.DEMON_1].prefstring[0] then
+        insert_preference(unit,1,list_of_creatures.DEMON_1)
+    end
+    if #df.global.world.poetic_forms.all then
```

Comment:
This doesn't do what you expect, because Lua is weird and treats `0` as `true`:
```lua
> if 0 then print('wat') end
wat
```
I'll add an explicit `> 0` check to these